src/app_model/backends/qt/_qkeymap.py

```diff
@@ -18,13 +18,26 @@
 QMETA = Qt.KeyboardModifier.MetaModifier
 QCTRL = Qt.KeyboardModifier.ControlModifier
 
-MAC = OperatingSystem.current().is_mac
-_QMOD_LOOKUP: Dict[str, Qt.KeyboardModifier] = {
-    "ctrl": QMETA if MAC else QCTRL,
-    "shift": Qt.KeyboardModifier.ShiftModifier,
-    "alt": Qt.KeyboardModifier.AltModifier,
-    "meta": QCTRL if MAC else QMETA,
-}
+
+def mac_ctrl_meta_swapped() -> bool:
+    """Whether or not Qt has swapped ctrl and meta for Macs."""
+    if not OperatingSystem.current().is_mac:
+        return False
+
+    app = QCoreApplication.instance()
+    if app is None:
+        return False
+
+    return not app.testAttribute(Qt.AA_MacDontSwapCtrlAndMeta)
+
+
+def _get_qmod_lookup() -> Dict[str, Qt.KeyboardModifier]:
```

Comment:
It would be nice if this didn't have to run on every key press, particularly on non mac systems.  Can you refactor this so that the meta_swapped function is called at most once per key press on a mac, and never on a non mac?  (Non mac shouldn't need anything besides a hash map lookup)

Reply:
changed this in two places, lmk if that's what you wanted or not

Reply:
works for me.  could later possible do two function definitions and safe the conditional on non mac platforms altogether

Reply:
i considered that, if we want to optimize this much tho we might as well be using cython/numba

Reply:
not sure I agree with that.  it's much easier to just avoid a python function call in the first place than to put it into cython/numba?  anyway, do what you like here.  i'm fine with it

Reply:
i meant more generally if we want to apply the mentality of "this is keypress stuff that gets called a lot let's make it as efficient as possible" the best way would be cython/numba

Reply:
another way i could put it is if optimization is that much more important than readability, we might as well use compilers